### PR TITLE
use process.stdout.write on server instead of fs.write

### PR DESCRIPTION
**Pull request by vladima**
fixes VSCode regression introduced in #5354.
pinging @zhengbli, can you check if this solution still solves #2758 and https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/379

### CODE REVIEW

File: src/server/server.ts [hunk 1]
```diff
@@ -156,36 +156,24 @@ namespace ts.server {
...
-                if (pending.length > 0) {
-                    drain();
+    let canWrite = true;
+    function writeMessage(s: string) {
+        if (!canWrite) {
+            pending.push(s);
+        }
+        else {
+            canWrite = false;
+            process.stdout.write(new Buffer(s, "utf8"), () => {
```
Comment (DanielRosenwasser) [inline src/server/server.ts]: Do you need to create a new buffer each time or can you avoid allocating one per call?

Comment (DanielRosenwasser) [inline src/server/server.ts]: Consider also moving the callback out.

Comment (vladima) [inline src/server/server.ts]: this is the common practice and standard buffer API. In theory we can pool then though in practice I'll start doing this only if I have convincing numbers

**Comment by mhegazy** [comment]
:+1:

**Comment by mhegazy** [comment]
we will need to port this to release-1.7

**Comment by zhengbli** [comment]
Just curious, what is the VSCode regression?

**Comment by vladima** [comment]
@zhengbli code in VSCode that read data from the stdout of the `tsserver` process stopped working after `tsserver` was switched to the asynchronous writing. The same code worked perfectly fine with the old implementation and with the proposed fix - tested on Windows and Linux machines

**Comment by zhengbli** [comment]
@vladima got it. Thanks for the explanation!

**Comment by mhegazy** [comment]
let's get this in first as this is blocking VSCode, and then we can update the fix if the original issue was not fixed.

**Comment by zhengbli** [comment]
reasonable. If this fix works for vscode on windows, mac, and Linux then :+1:

**Comment by zhengbli** [comment]
Confirmed the fix works on both Mac and Windows for sublime text.

**Pull request merged**